components/brave_wallet/browser/swap_service.cc

```diff
@@ -113,15 +180,28 @@ GURL Append0xSwapParams(const GURL& swap_url,
   if (!params.sell_token.empty()) {
     url = net::AppendQueryParameter(url, "sellToken", params.sell_token);
   }
-  url =
-      net::AppendQueryParameter(url, "buyTokenPercentageFee", GetFee(chain_id));
+
+  auto brave_swap_fee_params = brave_wallet::mojom::BraveSwapFeeParams::New();
+  brave_swap_fee_params->chain_id = chain_id;
+  brave_swap_fee_params->taker = params.taker_address;
+  brave_swap_fee_params->input_token = params.sell_token;
+  brave_swap_fee_params->output_token = params.buy_token;
+  const auto& brave_fee = GetBraveFeeInternal(std::move(brave_swap_fee_params));
+  double effective_fee_pct;
+  if (brave_fee &&
+      base::StringToDouble(brave_fee->effective_fee_pct, &effective_fee_pct) &&
+      effective_fee_pct > 0) {
+    url = net::AppendQueryParameter(url, "buyTokenPercentageFee",
+                                    brave_fee->fee_param);
+  }
```

Comment:
It is not clear why we are checking one field, but adding to a query another one. 
Needs comments or so.

Reply:
It should be clearer now with https://github.com/brave/brave-core/pull/19834/commits/0eed8dc23e53066c409f28b9cf4ebdb37b206405:

```cpp
if (brave_fee && brave_fee->has_brave_fee) {
  url = net::AppendQueryParameter(url, "feeBps", brave_fee->fee_param);
}
```